opentelemetry-instrumentation/src/opentelemetry/instrumentation/auto_instrumentation/sitecustomize.py

```diff
@@ -96,7 +113,14 @@ def _load_configurators():
             )
             continue
         try:
-            entry_point.load()().configure()  # type: ignore
+            configurator: BaseConfigurator = entry_point.load()()
+            if not isinstance(configurator, BaseConfigurator):
```

Comment:
Do we want to enforce nominal typing at runtime here? I don't see this adding a ton of benefit but these things keep adding up to the "startup" cost of the SDK which can be critical in environments like AWS Lambda with really low resources.

Reply:
If we consider startup costs should we remove the type check for `BaseDistro`?

https://github.com/open-telemetry/opentelemetry-python-contrib/blob/753e22896a11cadf28c4f64acdfcfd73350d7d90/opentelemetry-instrumentation/src/opentelemetry/instrumentation/auto_instrumentation/sitecustomize.py#L38-L42

I guess it has the benefit of checking a Distro and we know that Distro's have Configurators. I'll remove this but let me know if you think we should remove the `BaseDistro` check too.

Reply:
I'm not sure. What benefit do we get from this message? it's also logging on debug level but we are paying the cost of isinstance always. i personally feel we should remove this. This isn't something that is going to be very helpful to end users IMO. If anyone publishes a package with a distro entrypoint that doesn't satisfy the interface, an attribute/type error would probably be good enough I think

Reply:
Okay I agree with you! I will remove it since I'm changing those lines anyways.